pandas/core/reshape/tile.py

```diff
@@ -359,10 +360,16 @@ def _coerce_to_type(x):
     elif is_timedelta64_dtype(x):
         x = to_timedelta(x)
         dtype = np.timedelta64
+    elif is_bool_dtype(x):
+        dtype = x.dtype
 
     if dtype is not None:
-        # GH 19768: force NaT to NaN during integer conversion
-        x = np.where(x.notna(), x.view(np.int64), np.nan)
+        if is_bool_dtype(x):
+            # GH 20303: coerce bool to int
+            x = np.where(np.isnan(x), np.nan, x.astype(int))
+        else:
+            # GH 19768: force NaT to NaN during integer conversion
+            x = np.where(x.notna(), x.view(np.int64), np.nan)
```

Comment:
not sure you need to change any of this here; you could also simply do ``x = x.astype(int)`` above as well

Reply:
Adding `x = x.astype(int)` right after the `elif is_bool_dtype` check works if the argument is a `Series`. Otherwise, it throws an `AttributeError` when trying to call `notna` on a `ndarray`. 

It also seems to fail when converting a `ndarray` to `Series`, calling `astype(int)` and passing it as an argument (should I also test for this?). 

We could add a new `if` statement checking for `is_bool_dtype` after the `x = np.where(x.notna(), x.view(np.int64), np.nan)` line. Although I think I might be overthinking this.